Indicators/HilbertTransformIndicator.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Indicators;
+
+/// <summary>
+/// This indicator computes the Hilbert Transform Indicator by John Ehlers.
+/// By using present and prior price differences, and some feedback, price values are split into their complex number components
+/// of real (inPhase) and imaginary (quadrature) parts.
+/// http://www.technicalanalysis.org.uk/moving-averages/Ehle.pdf
+/// </summary>
+public class HilbertTransformIndicator : Indicator, IIndicatorWarmUpPeriodProvider
+{
+    private const int Quad2Length = 2;
+    private const int InPhase3Length = 3;
+
+    private readonly IndicatorBase<IndicatorDataPoint> _input;
+    private readonly IndicatorBase<IndicatorDataPoint> _prev;
+    private readonly IndicatorBase<IndicatorDataPoint> _v1;
+    private readonly IndicatorBase<IndicatorDataPoint> _v2;
+    private readonly IndicatorBase<IndicatorDataPoint> _v4;
+    private readonly Queue<IndicatorDataPoint> _inPhase3 = new();
+    private readonly Queue<IndicatorDataPoint> _quad2 = new();
+
+    private int updatesCount = 0;
+    private readonly int _inPhase3WarmUpPeriod;
+    private readonly int _quad2WarmUpPeriod;
+
+    /// <summary>
+    /// Real (inPhase) part of complex number component of price values
+    /// </summary>
+    public IndicatorBase<IndicatorDataPoint> InPhase { get; }
+
+    /// <summary>
+    /// Imaginary (quadrature) part of complex number component of price values
+    /// </summary>
+    public IndicatorBase<IndicatorDataPoint> Quad { get; }
+
+    /// <summary>
+    /// Creates a new Hilbert Transform indicator
+    /// </summary>
+    /// <param name="name">The name of this indicator</param>
+    /// <param name="length">The length of the FIR filter used in the calculation of the Hilbert Transform.
+    /// This parameter determines the number of filter coefficients in the FIR filter.</param>
+    /// <param name="iMult">The multiplication factor used in the calculation of the in-phase component
+    /// of the Hilbert Transform. This parameter adjusts the sensitivity and responsiveness of
+    /// the transform to changes in the input signal.</param>
+    /// <param name="qMult">The multiplication factor used in the calculation of the quadrature component of
+    /// the Hilbert Transform. This parameter also adjusts the sensitivity and responsiveness of the
+    /// transform to changes in the input signal.</param>
+    public HilbertTransformIndicator(string name, int length, decimal iMult, decimal qMult)
```

Comment:
Nit/style. In Lean, we avoid Hungarian notation or abbreviation, so `iMult` should be `inPhaseMultiplicationFactor` or `inPhaseMultiplier`. Same for `qMult`.

Reply:
fixed by 67cbc51ec4f762a4ac89b19c12a9576408995d98